java/bulk-export/bulk-export-core/src/test/java/sleeper/bulkexport/model/BulkExportLeafPartitionQuerySerDeTest.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+    @Test
+    public void shouldThrowExceptionNoLeafPartitionId() throws IOException {
+        // Given
+        String tableId = "t-id";
+        tableProperties.set(TABLE_ID, tableId);
+        BulkExportLeafPartitionQuerySerDe querySerDe = generateQuerySerDeFromTableProperties(schema);
+
+        String json = readFileToString(
+                "src/test/resources/BulkExportLeafPartitionQuerySerDeTest.shouldThrowExceptionNoLeafPartitionId.json");
+
+        // When & Then
+        assertThatThrownBy(() -> querySerDe.fromJson(json))
+                .isInstanceOf(
+                        BulkExportQueryValidationException.class)
+                .hasMessage("Query validation failed for export \"e-id\": leafPartitionId field must be provided");
+    }
+
+    @Test
+    public void shouldThrowExceptionNoPartitionRegion() throws IOException {
+        // Given
+        String tableId = "t-id";
+        tableProperties.set(TABLE_ID, tableId);
+        BulkExportLeafPartitionQuerySerDe querySerDe = generateQuerySerDeFromTableProperties(schema);
+
+        String json = readFileToString(
+                "src/test/resources/BulkExportLeafPartitionQuerySerDeTest.shouldThrowExceptionNoPartitionRegion.json");
+
+        // When & Then
+        assertThatThrownBy(() -> querySerDe.fromJson(json))
+                .isInstanceOf(
+                        BulkExportQueryValidationException.class)
+                .hasMessage("Query validation failed for export \"e-id\": partitionRegion field must be provided");
+    }
+
+    @Test
+    public void shouldThrowExceptionNoFiles() throws IOException {
+        // Given
+        String tableId = "t-id";
+        tableProperties.set(TABLE_ID, tableId);
+        BulkExportLeafPartitionQuerySerDe querySerDe = generateQuerySerDeFromTableProperties(schema);
+
+        String json = readFileToString(
+                "src/test/resources/BulkExportLeafPartitionQuerySerDeTest.shouldThrowExceptionNoFiles.json");
+
+        // When & Then
+        assertThatThrownBy(() -> querySerDe.fromJson(json))
+                .isInstanceOf(
+                        BulkExportQueryValidationException.class)
+                .hasMessage("Query validation failed for export \"e-id\": files field must be provided");
+    }
+
+    private BulkExportLeafPartitionQuerySerDe generateQuerySerDeFromTableProperties(Schema schema) {
+        tableProperties.setSchema(schema);
+        return new BulkExportLeafPartitionQuerySerDe(new FixedTablePropertiesProvider(tableProperties));
+    }
+
+    private BulkExportLeafPartitionQuerySerDe generateQuerySerDeFromSchema(Schema schema) {
+        return new BulkExportLeafPartitionQuerySerDe(schema);
+    }
+
+    private String readFileToString(String path) throws IOException {
+        byte[] encoded = Files.readAllBytes(Paths.get(path));
+        return new String(encoded, StandardCharsets.UTF_8);
```

Comment:
This is relying on the current working directory, which seems a bit brittle. You can load resources with `CurrentClass.class.getClassLoader().getResourceAsStream(path)`.

Reply:
Updated to use getClassLoader()